packages/ckeditor5-paste-from-office/src/filters/removestyleblock.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module paste-from-office/filters/removestyleblock
+ */
+
+import type { UpcastWriter, ViewDocumentFragment } from 'ckeditor5/src/engine';
+
+/**
+ * Removes `<style>` block added by Google Sheets to a copied content.
+ *
+ * @param documentFragment element `data.content` obtained from clipboard
+ */
+export default function removeStyleBlock( documentFragment: ViewDocumentFragment, writer: UpcastWriter ): void {
+	for ( const child of documentFragment.getChildren() ) {
+		if ( child.is( 'element', 'style' ) ) {
+			writer.remove( child );
```

Comment:
This changes the list while iterating over it? Does the iterator point to valid position?

Please add tests for more than one style elements (consecutive and not).

Probably you should iterate over `Array.from( documentFragment.getChildren() )`.

Reply:
Done :heavy_check_mark: